frontend/src/pages/user/UserProfileViewPage.js

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from "react";
+import {
+  viewUserProfile,
+  checkIfFollowedUser,
+  followUser,
+  unfollowUser,
+  getUser,
+  checkFollowersAndFollowing,
+  showLearnedWord,
+} from "../../api/api";
+import { Link } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import { useToast } from "../../hooks/useToast";
+
+export default function UserProfileViewPage() {
+  const [user, setUser] = useState([]);
+  const [currentUser, setCurrentUser] = useState([]);
+  const [followersFollowings, setFollowersFollowings] = useState([]);
+  const [isFollowed, setIsFollowed] = useState(false);
+  const [learned, setLearned] = useState([]);
+  const { userId } = useParams();
+  const { showToast } = useToast();
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      const learnedResponse = await showLearnedWord(userId);
+      setLearned(learnedResponse);
+
+      const response = await viewUserProfile(userId);
+      setUser(response.data.data);
+
+      const checkFollowed = await checkIfFollowedUser(userId);
+      setIsFollowed(checkFollowed.data.followed);
+
+      const current = await getUser();
+      setCurrentUser(current.data.data);
+
+      const followFollowingCount = await checkFollowersAndFollowing(userId);
+      setFollowersFollowings(followFollowingCount.data);
+    };
+
+    fetchUser();
+  }, []);
+
+  const totalWordLearned = learned.map((learn) => {
+    return learn.user_id.length;
+  });
+
+  const handleFollow = async () => {
+    const follow = await followUser(userId);
+
+    setIsFollowed(follow.data);
+    showToast("success", follow.data.message);
+  };
+  const handleUnfollow = async () => {
+    const unfollow = await unfollowUser(userId);
+
+    setIsFollowed(false);
+    showToast("warning", unfollow.data.message);
+  };
```

Comment:
This can be done in 1 api
https://stackoverflow.com/questions/54982301/how-to-build-an-api-to-follow-and-unfollow-user-in-laravel-with-postman